### ISSUE TITLE
Avoid creating a new HttpClient on every request

### ISSUE THREAD

**Issue by whitfin** [lake54/groupsio-api-java#2]
Seems unnecessary to keep creating one, as we can just keep the client instance in the main `GroupsIOApiClient` and allow it to be closed. I'll gladly contribute this back if you're willing to accept the changes :)

**Comment by lake54** [comment]
Yep, happy for any/all PRs on this project. Haven't had much time recently to finish the new/updated calls off, and so other things like a proper review of things like that haven't taken place yet. If you're happy to do the work that would be great :-)
